processmanager/processinfo.go

```diff
@@ -88,11 +89,25 @@ func (pm *ProcessManager) updatePidInformation(pid libpf.PID, m *Mapping) (bool,
 		if name, err := os.ReadFile(fmt.Sprintf("/prod/%d/comm", pid)); err == nil {
 			processName = string(name)
 		}
+
+		envVarMap := make(map[string]string)
```

Comment:
To avoid multiple small reallcoations, `envVarMap` could be allocated to the expected length.
```suggestion
		envVarMap := make(map[string]string, len(pm.includeEnvVars))
```